### PR TITLE
refactor: android 23 compatibility

### PR DESCRIPTION
**Pull request by minh-rakuten**
# Description
SDK:
Resolve phone scheme detection issue on Android 6 by replacing the current`shouldOverrideUrlLoading`. Have tested on other OS and found no problem as well.

Sample app:
Add UUID length check.

## Links
MINI-1614
MINI-1620

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebViewClient.kt [hunk 1]
```diff
@@ -28,17 +28,16 @@ internal class MiniAppWebViewClient(
         return response
     }
 
-    override fun shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest): Boolean {
-        var shouldCancelLoading = super.shouldOverrideUrlLoading(view, request)
-        if (request.url != null) {
-            val requestUrl = request.url.toString()
-            if (requestUrl.startsWith("tel:")) {
-                openPhoneDialer(requestUrl)
+    override fun shouldOverrideUrlLoading(view: WebView?, url: String?): Boolean {
```
Comment (khairul-alam-licon) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebViewClient.kt:31]: According to the [official doc of WebViewClient](https://developer.android.com/reference/android/webkit/WebViewClient), this method has been deprecated in API level `24`.  But, this is the working function below of `24`, I think, it's a bug itself inside Android SDK.. 🤔 

My idea is using latest `shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest)` for API level `24+`, and using `shouldOverrideUrlLoading(view: WebView?, url: String?)` for below than `24`. 

So, we can refactor this part as following:

```
    @SuppressWarnings("deprecation")
    override fun shouldOverrideUrlLoading(view: WebView?, url: String?): Boolean {
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N)
            return overrideUrlLoading(url.toString(), super.shouldOverrideUrlLoading(view, url))

        return false
    }

    @RequiresApi(Build.VERSION_CODES.N)
    override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest): Boolean {
        return overrideUrlLoading(request.url.toString(), super.shouldOverrideUrlLoading(view, request))
    }

    private fun overrideUrlLoading(requestUrl: String, shouldOverride: Boolean): Boolean {
        var shouldCancelLoading = shouldOverride
        if (requestUrl.startsWith("tel:")) {
            openPhoneDialer(requestUrl)
            shouldCancelLoading = true
        } else if (....) {
           ...
        }
        return shouldCancelLoading
    }
```

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/helper/UUIDHelper.kt [hunk 2]
```diff
@@ -3,9 +3,10 @@ package com.rakuten.tech.mobile.testapp.helper
 import java.lang.IllegalArgumentException
 import java.util.UUID
 
+private const val UUID_LENGTH = 36
```
Comment (khairul-alam-licon) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/helper/UUIDHelper.kt:6]: 👍

**Review by khairul-alam-licon — APPROVED** [review]
Thanks for adding the fix, I left one idea.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebViewClient.kt (hunk 1 above)
Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebViewClient.kt:31]: It's a good idea but I think we can omit this version checking by using only one `shouldOverrideUrlLoading`. I tested and it's working fine on other android versions.
We may update our sdk to 24 as minimum later so let's keep this fix simple.

Comment (khairul-alam-licon) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebViewClient.kt:31]: Yes, we need to use `override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest)` when we can update the sdk to 24 as minimum level. For now, the current change is also accepted, so it's approved from my side 👍

Comment (corycaywood) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebViewClient.kt:31]: @minh-rakuten Can you add a `TODO` here so we remember to update this in the future?

**Review by corycaywood — APPROVED** [review]
Looks okay 👍

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebViewClient.kt (hunk 1 above)
Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebViewClient.kt:32]: ToDo has been added.

**Pull request merged**